src/client_code/client.c

```diff
@@ -58,42 +58,71 @@ int main(int argc, char* argv[]) {
   // Send the local TCP connection info to the server
   sendTcpAddress(serverAddress, tcpAddress, debugFlag);
 
+  fd_set read_fds;
+
+  char* message = malloc(1000);
+  struct sockaddr_in address;
+
   // Constantly check user input
   while(1) {
-    // Get user input and store in userInput buffer
-    char* userInput = malloc(USER_INPUT_BUFFER_LENGTH);
-    getUserInput(userInput);
 
-    // User just pressed return
-    if (strlen(userInput) == 0) {  
-      continue;
-    }
+    // Use select to handle user input and server messages simultaneously
+    FD_ZERO(&read_fds);
+    FD_SET(0, &read_fds);  // 0 is stdin (for user input)
+    FD_SET(udpSocketDescriptor, &read_fds);  // The socket for receiving server messages
 
-    if (checkStringForCommand(userInput) == 0) {            // User entered plain text message
-      sendUdpMessage(serverAddress, userInput, debugFlag);  // Send user input via UDP
-      continue;
-    }
+    int activity = select(udpSocketDescriptor + 1, &read_fds, NULL, NULL, NULL);
 
-    if (checkForValidCommand(userInput) == 0) {             // Not a recognized command or an invalid file name
-      printf("Please enter a valid command:\n");
-      printf("%%put to send a file to the server\n");
-      printf("%%get to request a file from the server\n");
-      continue;
+    if (activity < 0 && errno != EINTR) {
+        perror("select error");
     }
 
-    char* fileName = malloc(FILE_NAME_SIZE);
-    fileNameFromCommand(userInput, fileName);               // Extract the file name from the user input
-    
-    if (strncmp(userInput, "%put ", 5) == 0) {              // Put command
-      sendUdpMessage(serverAddress, userInput, debugFlag);  // Send user input(command) via UDP
-      putCommand(fileName);                                 // Send the file
-    }
-    else {                                                  // Get command
-      sendUdpMessage(serverAddress, userInput, debugFlag);  // Send user input(command) via UDP
-      if (getCommand(fileName) == -1) {                     // Receive the file
-        printf("Failed to get file\n");
+    if (FD_ISSET(0, &read_fds)) {
+      // Get user input and store in userInput buffer
+      char* userInput = malloc(USER_INPUT_BUFFER_LENGTH);
+      getUserInput(userInput);
```

Comment:
This may not be cleared correctly or experiencing a "one off" error where one or two characters are being left in the buffer when the new user input is taken in